crates/starknet-server/tests/test_simulate_transactions.rs

```diff
@@ -0,0 +1,403 @@
+            ERC20_CONTRACT_ADDRESS.to_lowercase()
+        );
+        assert_eq!(
+            skip_validation_trace["constructor_invocation"]["contract_address"].as_str().unwrap(),
+            account_address_hex
+        );
+
+        assert_fee_in_resp_greater(resp_no_flags, resp_skip_validation);
+
+        // skipped validation and fee charging (everything)
+        let params_skip_everything = get_params(&["SKIP_VALIDATE", "SKIP_FEE_CHARGE"]);
+        let resp_skip_everything = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_skip_everything)
+            .await["result"][0];
+        let skip_everything_trace = &resp_skip_everything["transaction_trace"];
+        assert!(skip_everything_trace["validate_invocation"].as_object().is_none());
+        assert!(skip_everything_trace["fee_transfer_invocation"].as_object().is_none());
+        assert_eq!(
+            skip_everything_trace["constructor_invocation"]["contract_address"].as_str().unwrap(),
+            account_address_hex
+        );
+    }
+
+    #[tokio::test]
+    async fn simulate_invoke() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        // get account
+        let (signer, account_address) = get_predeployed_account_props();
+        let account = Arc::new(SingleOwnerAccount::new(
+            devnet.clone_provider(),
+            signer,
+            account_address,
+            CHAIN_ID,
+            ExecutionEncoding::Legacy,
+        ));
+
+        // get class
+        let contract_json = dummy_cairo_0_contract_class();
+        let contract_artifact: Arc<LegacyContractClass> =
+            Arc::new(serde_json::from_value(contract_json.inner).unwrap());
+        let class_hash = contract_artifact.class_hash().unwrap();
+
+        // declare class
+        let declaration_result =
+            account.declare_legacy(contract_artifact.clone()).send().await.unwrap();
+        assert_eq!(declaration_result.class_hash, class_hash);
+
+        // deploy instance of class
+        let contract_factory = ContractFactory::new(class_hash, account.clone());
+        let salt = FieldElement::from_hex_be("0x123").unwrap();
+        let constructor_calldata = vec![];
+        let contract_address = get_udc_deployed_address(
+            salt,
+            class_hash,
+            &UdcUniqueness::NotUnique,
+            &constructor_calldata,
+        );
+        contract_factory.deploy(constructor_calldata, salt, false).send().await.unwrap();
+
+        // prepare the call used in simulation
+        let increase_amount = FieldElement::from(100u128);
+        let invoke_calls = vec![Call {
+            to: contract_address,
+            selector: get_selector_from_name("increase_balance").unwrap(),
+            calldata: vec![increase_amount],
+        }];
+
+        // TODO fails if max_fee too low, can be used to test reverted case
```

Comment:
this TODO will stay or it will be handled?

Reply:
As noted in the PR description, delegated to https://github.com/0xSpaceShard/starknet-devnet-rs/issues/192